### DISCUSSION TITLE
Does shouldUseNativeValidation function using useController/Controller?

### DISCUSSION THREAD

**Discussion by d-henn** [react-hook-form/react-hook-form#8891] [General] [answered]
## Question 

Does the `shouldUseNativeValidation ` prop function properly with `useController`/`Controller`?

The documentation lends me to believe it doesn't work as it's stated under this [section](https://www.react-hook-form.com/api/useform/#resolver) ~ "`This feature only works for register API, not useController/Controller.`"

**_However_** I was able to implement this using the `Controller`, am I missing something important with this prop? Or does the documentation just need to be updated for that section?

Thanks! 

>Btw, incredible library!

**Comment by Moshyfawn** [comment, marked as answer]
The docs are correct: when `shouldUseNativeValidation: true`, the `register` method returns the native validation attributes defined on the second `register` options argument, like `required: true`, `min: 10`, etc. The `Controller`'s `field` prop returned from it's `render` method  callback always returns the field values and callbacks like `value`, `onChange`, `ref` etc

**Reply by Moshyfawn** [reply]
Thx for the kind words about the lib, btw ;)

**Reply by d-henn** [reply]
I see, I suppose the `invalid`/`valid` work, but there could be some things missing from the returns. Thanks :)
